unstract/connectors/src/unstract/connectors/filesystems/azure_cloud_storage/azure_cloud_storage.py

```diff
@@ -72,3 +74,36 @@ def test_credentials(self) -> bool:
                 f"Error from Azure Cloud Storage while testing connection: {str(e)}"
             ) from e
         return True
+
+    def upload_file_to_storage(self, source_path: str, destination_path: str) -> None:
+        """Method to upload filepath from tool to destination connector
+        directory.
+
+        Args:
+            source_path (str): source file path from tool
+            destination_path (str): destination azure directory file path
+
+        Raises:
+            AzureHttpError: returns error for invalid directory
+        """
+        normalized_path = os.path.normpath(destination_path)
+        fs = self.get_fsspec_fs()
+        try:
+            with open(source_path, "rb") as source_file:
+                fs.write_bytes(normalized_path, source_file.read())
+        except AzureException.HttpResponseError as e:
+            self.raise_http_exception(e=e, path=normalized_path)
+
+    def raise_http_exception(
+        self, e: AzureException.HttpResponseError, path: str
+    ) -> AzureHttpError:
+        user_message = f"Error from Azure Cloud Storage connector. {e.reason} "
+        if e.status_code == 400:
+            user_message = (
+                f"Error from Azure Cloud Storage connector. "
+                f"Invalid resource name for path '{path}'. {e.reason}"
+            )
```

Comment:
@kirtimanmishrazipstack what if this 400 status code is for an invalid value of a different field? Won't the `e.reason` be different in that case?

Reply:
@chandrasekharan-zipstack I tried using

if e.error_code.name == "INVALID_RESOURCE_NAME": # type: ignore
but it was throwing `Pylance` issue. So used status_code 400.

This is more declariative but has pylance issue

Reply:
handled @chandrasekharan-zipstack